### PR TITLE
feat: add chunkFormat & chunkLoading & enabledChunkLoadingTypes config

### PR DESCRIPTION
**Pull request by underfin**
## Related issue (if exists)

closed at https://github.com/web-infra-dev/rspack/issues/2672

## Summary

- add chunkFormat & chunkLoading & enabledChunkLoadingTypes config
- fix globalObject at runtime

### <samp>🤖 Generated by Copilot at 61edfd3</samp>

Added and improved chunk format and loading options for `rspack` output bundles. Fixed a bug with the UMD library target and the runtime chunk. Added tests for the new features and the bug fix. Refactored some code in `rspack_binding_options` and `config` modules.

<details open=true>
  <summary><h2>Walkthrough</h2></summary>

### <samp>🤖 Generated by Copilot at 61edfd3</samp>

*  Remove unused `TargetPlatform` import from `options/mod.rs` ([link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-6112b39f2c1def8a0f89824413cfd6936f187aa6ca3712e1e6aeb4d34f67ae41L7-R7))
*  Replace `match` expression with more flexible output options in `options/mod.rs` ([link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-6112b39f2c1def8a0f89824413cfd6936f187aa6ca3712e1e6aeb4d34f67ae41L119-L134))
*  Import `rspack_error` crate and add fields and methods for output options in `options/raw_output.rs` ([link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-ef5856a7fec8410383d58cc835932b263f5df692fd8d31fb1ea236182a162b00R6), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-ef5856a7fec8410383d58cc835932b263f5df692fd8d31fb1ea236182a162b00R124-R126), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-ef5856a7fec8410383d58cc835932b263f5df692fd8d31fb1ea236182a162b00R132-R134), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-ef5856a7fec8410383d58cc835932b263f5df692fd8d31fb1ea236182a162b00R171-R220))
*  Convert and normalize output options in `config/adapter.ts` and `config/normalization.ts` ([link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-5f5bb1aeb2a977ff30c5ab822c79f09b76476c503462939c69beb9e4d7552123L167-R169), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-5f5bb1aeb2a977ff30c5ab822c79f09b76476c503462939c69beb9e4d7552123L182-R185), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-55e0fbd74437dff4e251152f5d93efd7f5871cbad72a7959dddb6add8fbf60ccL81-R86), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-55e0fbd74437dff4e251152f5d93efd7f5871cbad72a7959dddb6add8fbf60ccL285-R298), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-78648f723799beccc94c4b2e506997d12d1ace8216aeb31c0731c37a6dce024aR53-R54), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-78648f723799beccc94c4b2e506997d12d1ace8216aeb31c0731c37a6dce024aR61-R63))
*  Fix indentation issues in `config/adapter.ts` ([link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-5f5bb1aeb2a977ff30c5ab822c79f09b76476c503462939c69beb9e4d7552123L208-R215), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-5f5bb1aeb2a977ff30c5ab822c79f09b76476c503462939c69beb9e4d7552123L219-R222), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-5f5bb1aeb2a977ff30c5ab822c79f09b76476c503462939c69beb9e4d7552123L226-R234), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-5f5bb1aeb2a977ff30c5ab822c79f09b76476c503462939c69beb9e4d7552123L272-R279), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-5f5bb1aeb2a977ff30c5ab822c79f09b76476c503462939c69beb9e4d7552123L380-R384), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-5f5bb1aeb2a977ff30c5ab822c79f09b76476c503462939c69beb9e4d7552123L400-R413), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-5f5bb1aeb2a977ff30c5ab822c79f09b76476c503462939c69beb9e4d7552123L431-R436), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-5f5bb1aeb2a977ff30c5ab822c79f09b76476c503462939c69beb9e4d7552123L453-R457))
*  Add definitions and fields for output options in `config/schema.js` and `config/types.ts` ([link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-d7e2c50a4845569f35f387513dc5a6e542e9d5dde107294fd55c5ff4e8a72c88R65-R100), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-d7e2c50a4845569f35f387513dc5a6e542e9d5dde107294fd55c5ff4e8a72c88R141-R148), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-d7e2c50a4845569f35f387513dc5a6e542e9d5dde107294fd55c5ff4e8a72c88R1042-R1050), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-790d97d1c7c2bfbbe7720a38355ec1e0d462206a5ddb9826c6aa8bff7090ce0fR138-R140), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-790d97d1c7c2bfbbe7720a38355ec1e0d462206a5ddb9826c6aa8bff7090ce0fR234-R236))
*  Update and add test cases for output options and UMD library target in `tests/` ([link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-68435635c7abf06035e99ade8f46eeda6ad7614810533989afc5a1e324eb9db9R262-R263), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-68435635c7abf06035e99ade8f46eeda6ad7614810533989afc5a1e324eb9db9R269-R271), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-68435635c7abf06035e99ade8f46eeda6ad7614810533989afc5a1e324eb9db9R858-R865), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-68435635c7abf06035e99ade8f46eeda6ad7614810533989afc5a1e324eb9db9R914-R919), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-68435635c7abf06035e99ade8f46eeda6ad7614810533989afc5a1e324eb9db9R942-R949), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-68435635c7abf06035e99ade8f46eeda6ad7614810533989afc5a1e324eb9db9R1009-R1016), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-68435635c7abf06035e99ade8f46eeda6ad7614810533989afc5a1e324eb9db9R1312-R1313), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-ecdc6c34607483a3fa62a5ff51e6cbd6fbacb7ca9e1f6302ebe645a18a986cd9R1-R5), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-601deec5a45b08a96d1f4c59c8db14fdd566335c8e8e1190b5009beb6bae6fe1R1-R9), [link](https://github.com/web-infra-dev/rspack/pull/2792/files?diff=unified&w=0#diff-a44cd9eceb1119aef67cb1496f7ac9b3f5c6ee5ffc9d054cd3ab5d8d41c690bfR1-R19))

</details>

### CODE REVIEW

File: packages/rspack/src/config/defaults.ts [hunk 1]
```diff
@@ -341,6 +354,78 @@ const applyOutputDefaults = (
 	if (output.library) {
 		F(output.library, "type", () => (output.module ? "module" : "var"));
 	}
+	F(output, "chunkFormat", () => {
+		if (tp) {
+			const helpMessage = isAffectedByBrowserslist
+				? "Make sure that your 'browserslist' includes only platforms that support these features or select an appropriate 'target' to allow selecting a chunk format by default. Alternatively specify the 'output.chunkFormat' directly."
+				: "Select an appropriate 'target' to allow selecting one by default, or specify the 'output.chunkFormat' directly.";
+			if (output.module) {
+				if (tp.dynamicImport) return "module";
```
Comment (hardfist) [inline packages/rspack/src/config/defaults.ts:363]: not support module now?

File: packages/rspack/src/config/schema.js [hunk 2]
```diff
@@ -62,6 +62,42 @@ module.exports = {
 				}
 			]
 		},
+		ChunkFormat: {
+			description:
+				"The format of chunks (formats included by default are 'array-push' (web/WebWorker), 'commonjs' (node.js), 'module' (ESM), but others might be added by plugins).",
+			anyOf: [
+				{
+					enum: ["array-push", "commonjs", "module", false]
```
Comment (hardfist) [inline packages/rspack/src/config/schema.js:70]: module is not supported now

Comment (underfin) [reply packages/rspack/src/config/schema.js:70]: I will add this at next pr.

File: packages/rspack/src/config/defaults.ts (hunk 1 above)
Comment (underfin) [reply packages/rspack/src/config/defaults.ts:363]: I will add this at next pr.

**Review by hyf0 — COMMENTED** [review]
Looking at the code, the PR seems to be a feature rather than a bug fix.

**Review by hardfist — APPROVED** [review]

**Pull request merged**